pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -305,17 +326,37 @@ func syncClaim(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCli
 			return fmt.Errorf("Unexpected error getting persistent volume: %v\n", err)
 		}
 
+		// TODO:  Can this be removed? The ref will not be nil for volumes bound by this controller.
 		if volume.Spec.ClaimRef == nil {
 			glog.V(5).Infof("Rebuilding bind on pv.Spec.ClaimRef\n")
 			claimRef, err := api.GetReference(claim)
 			if err != nil {
 				return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
 			}
-			volume.Spec.ClaimRef = claimRef
-			_, err = binderClient.UpdatePersistentVolume(volume)
+			clone, err := conversion.NewCloner().DeepCopy(volume)
 			if err != nil {
+				return fmt.Errorf("Error cloning pv: %v", err)
+			}
+			volumeClone, ok := clone.(*api.PersistentVolume)
+			if !ok {
+				return fmt.Errorf("Unexpected pv cast error : %v\n", volumeClone)
+			}
+			volumeClone.Spec.ClaimRef = claimRef
+			if updatedVolume, err := binderClient.UpdatePersistentVolume(volumeClone); err != nil {
 				return fmt.Errorf("Unexpected error saving PersistentVolume.Status: %+v", err)
+			} else {
+				volumeIndex.Update(updatedVolume)
+			}
+		}
+
+		claimKey := fmt.Sprintf("%s/%s", claim.Namespace, claim.Name)
+		claimRefKey := fmt.Sprintf("%s/%s", volume.Spec.ClaimRef.Namespace, volume.Spec.ClaimRef.Name)
+		if claimKey != claimRefKey {
```

Comment:
Is the intent to have a small optimization by not creating those 2 strings?  Your `if` clause would work equally as well as what is here.